pkg/operator/sync.go

```diff
@@ -126,6 +142,76 @@ func (optr *Operator) syncTerminationHandler(config *OperatorConfig) error {
 	return nil
 }
 
+func (optr *Operator) syncWebhookConfiguration(config *OperatorConfig) error {
+	if err := optr.syncValidatingWebhook(); err != nil {
+		return err
+	}
+
+	return optr.syncMutatingWebhook()
+}
+
+func (optr *Operator) syncValidatingWebhook() error {
+	client := optr.kubeClient.AdmissionregistrationV1().ValidatingWebhookConfigurations()
+	requiredHook := mapiv1.NewValidatingWebhookConfiguration()
+	getHook, err := optr.validatingWebhookLister.Get(requiredHook.Name)
+	if apierrors.IsNotFound(err) {
+		_, err = client.Create(context.Background(), requiredHook, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+		return nil
+	} else if err != nil {
+		return err
+	}
+	// Usage of DeepDerivative is needed, as there otherwise will be no full equality between
+	// the populated resource and the template, due to default values being already set on the former
+	// The downside - an additional check for the list lenght is needed, as it can't recognize webhook removal,
+	// if the order stays the same
+	if len(requiredHook.Webhooks) != len(getHook.Webhooks) ||
+		!equality.Semantic.DeepDerivative(requiredHook.Webhooks, getHook.Webhooks) ||
+		!equality.Semantic.DeepDerivative(requiredHook.Annotations, getHook.Annotations) {
+		updateHook := getHook.DeepCopy()
+		updateHook.Annotations = requiredHook.Annotations
+		updateHook.Webhooks = requiredHook.Webhooks
+		_, err = client.Update(context.Background(), updateHook, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (optr *Operator) syncMutatingWebhook() error {
+	client := optr.kubeClient.AdmissionregistrationV1().MutatingWebhookConfigurations()
+	requiredHook := mapiv1.NewMutatingWebhookConfiguration()
```

Comment:
What about calling this `currentHook`, it's a bit more obvious what it is I think. To me, `getHook` is a function name, because it sounds like an action